packages/in-app-notification-manager/src/services/inAppNotificationService.ts

```diff
@@ -0,0 +1,130 @@
+import { and, desc, eq, getTableColumns, ilike, inArray } from "drizzle-orm";
+import { drizzle } from "drizzle-orm/node-postgres";
+import {
+  AppContext,
+  createListResult,
+  escapeRegExp,
+  UIAuthData,
+  WithLogger,
+  withTotalCount,
+} from "pagopa-interop-commons";
+import { notification } from "pagopa-interop-in-app-notification-db-models";
+import {
+  fromNotificationSQL,
+  ListResult,
+  Notification,
+} from "pagopa-interop-models";
+import { notificationNotFound } from "../model/errors.js";
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function inAppNotificationServiceBuilder(
+  db: ReturnType<typeof drizzle>
+) {
+  return {
+    getNotifications: async (
+      q: string | undefined,
+      limit: number,
+      offset: number,
+      {
+        logger,
+        authData: { userId, organizationId },
+      }: WithLogger<AppContext<UIAuthData>>
+    ): Promise<ListResult<Notification>> => {
+      logger.info("Getting notifications");
+      const notifications = await db
+        .select(withTotalCount(getTableColumns(notification)))
+        .from(notification)
+        .where(
+          and(
+            eq(notification.userId, userId),
+            eq(notification.tenantId, organizationId),
+            q ? ilike(notification.body, `%${escapeRegExp(q)}%`) : undefined
+          )
+        )
+        .orderBy(desc(notification.createdAt))
+        .limit(limit)
+        .offset(offset);
+
+      return createListResult(
+        notifications.map(({ totalCount: _, ...n }) => fromNotificationSQL(n)),
+        notifications[0]?.totalCount ?? 0
+      );
+    },
+    markNotificationAsRead: async (
+      notificationId: string,
```

Comment:
Could we add a new `NotificationId` branded type for this new resource Id? And use it everywhere we have `notificationId` in this file